### ISSUE TITLE
Cant read DUCKDB after applying a using the 'dplyr'  dbWriteTable with a large size a 2nd time.

### ISSUE THREAD

**Issue by BertTijhuis** [cwida/duckdb#322]
Currently I was trying to use duckDB instead of SQLite3 as database, Reason is that I want to compare the both databases on speed.

But running the next code: 

````R
library(readr)
library(DBI)

db = 'imdb_duck.db'
file.remove(db)

mydb <- dbConnect(duckdb::duckdb(), db)

title.basics <- read_delim(
  "https://datasets.imdbws.com/title.basics.tsv.gz",
  "\t",
  escape_double = FALSE,
  col_types = cols(
    endYear = col_integer(),
    
    isAdult = col_logical(),
    runtimeMinutes = col_integer(),
    startYear = col_integer()
  ),
  na = c("", "\\N"),
  trim_ws = TRUE
)

dbWriteTable(mydb, "title_basics", title.basics)
rm(title.basics)
gc()

dbDisconnect(mydb)
````
The database is looking like this:

> imdb_duck.db         12KB
> imdb_duck.db.wal   612072 KB

When trying to open the database again the next message is appearing:

> mydb <- dbConnect(duckdb::duckdb(), db)
Error in duckdb_connection(drv, dbdir = dbdir, debug = debug) : 
  duckdb_startup_R: Failed to open database
> 

I've tried to use the `db_commit(mydb)`  directly after the dbWriteTable but this doesn't resolve the issue.

The same code works fine for an SQLite3 db.

Please help me (it is not really urgent)

**Comment by hannesmuehleisen** [comment]
Are you shutting down your R session in between? When doing that, this works fine for me. What we are indeed missing in the R package right now is a proper shutdown function.

**Comment by hannesmuehleisen** [comment]
See also #323

**Comment by BertTijhuis** [comment]
Indeed after shuting down the R session. I was able to connect to the database.

Thanks for you support explaining this behaviour

**Comment by edrubin** [comment]
After appending several times to a table, I'm having this same problem. The problem occurs even after shutting down the connection via `dbDisconnect(duck_con, shutdown = T)` and/or `duckdb::duckdb_shutdown(duckdb::duckdb())`. If I restart R, I still cannot connect to the database.

Is it necessary to entirely shutdown and restart R between connections to the database?

*Note:* The problem happens after appending several big-ish datasets to the same table (and disconnecting and reconnecting in between appends). 
- If I make the datasets very small (*e.g.*, 22 datasets of 1,000 observations each), I don't have the issue.
- After the first few appends, I can still connect. It's after the 7th or 8th append that things fall apart.

```R

# Setup ------------------------------------------------------------------------
  # Packages
  library(pacman)
  p_load(tidyverse, duckdb, data.table, magrittr, here)

# Build database ---------------------------------------------------------------
  # Looping over files, read from csv.bz2 files and write to DuckDB database
  for (i in 1987:2008) {
    # Print 'i'
    cat(i, "\n")
    # Read data
    tmp = paste0(
      "http://www.rdatasciencecases.org/Data/Airline/", i, ".csv.bz2"
    ) %>% fread()
    # Create connection (or reconnect)
    duck_con = dbConnect(duckdb::duckdb(), here("test-db"))
    # Write the file to the database
    dbWriteTable(
      conn = duck_con,
      name = "flights",
      value = tmp,
      append = T
    )
    # Disconnect
    dbDisconnect(duck_con, shutdown = T)
    duckdb::duckdb_shutdown(duckdb::duckdb())
    rm(tmp, duck_con)
    invisible(gc())
  }
```

*Note 2:* `RSQLite` works fine with the code (essentially just changing the driver and dropping the DuckDB-specific shutdown function.

**Comment by edrubin** [comment]
Quick follow up: I wrapped up script in a bash script so that R restarts in between appends. It still fails after the 8th file (1994).

**Comment by tyluRp** [comment]
I am also having this error pop up while trying working with duckdb and shiny.

**Comment by tyluRp** [comment]
I am running into this error quite a bit while using `duckdb` along with `pool` in a `shiny` app. Am I understanding correctly that this is happening because I am not shutting down the database properly? I have something like this in my `global.R` file:

```r
pool <- dbPool(
  drv = duckdb::duckdb(),
  dbdir = "data/db/db.duckdb"
)

onStop(function() {
  poolClose(pool)
})
```

**Comment by hannesmuehleisen** [comment]
The .wal transfer issue has been fixed in the 0.2.5, its just not on CRAN yet.

**Comment by MarcoFelipeKing** [comment]
I have downloaded 0.2.5 from the GitHub repository but still find the same problem. I have set up a simple example which I have uploaded to a server:

It will run once, but If I close it and open it again, it won't display the db. The only way I can get it to display a few rows of the DB a second time, is if I edit the .Rmd file and resave. Does this point to the same connection closing issue?

```
---
title: "Test"
output:
  flexdashboard::flex_dashboard:
    orientation: columns
    vertical_layout: fill
    runtime: shiny
---

```{r setup, include=FALSE}
library(flexdashboard)
library(shiny)
library(DBI)

con = dbConnect(duckdb::duckdb(), dbdir=":memory:", read_only=FALSE)
DBI::dbWriteTable(con, "mtcars", mtcars)

onStop(function(){
    DBI::dbDisconnect(con)
  })
```

Column {data-width=650}
-----------------------------------------------------------------------

### Chart A

```{r}
a<-dbGetQuery(con, 'SELECT * FROM mtcars LIMIT 5')
DT::renderDataTable(a)

```

```
